DMTN-148.tex

```diff
@@ -50,9 +50,276 @@
 % ADD CONTENT HERE
 % You can also use the \input command to include several content files.
 
+\section{Intro}
+
+The goal of this tech note is to provide a plan for the handling of
+calibration products in the Gen-3 Butler environment.  There are a
+large number of these products currently in use, but many of them do
+not have a clearly defined process to define, ingest, test, and
+deprecate them.  The lack of documentation on how these processes work
+has led to many implementations of similar paths, further complicating
+the issue.
+
+
+\section{The End State}
+
+The final state that we wish to get to has the following properties:
+
+\begin{itemize}
+\item All calibration products are either lsst.afw.image.Exposure
+  image data or an explicit calibration class in the \verb|ip_isr|
+  package.\footnote{This will require class reorganization to migrate
+    the Defects and Curve calibrations definitions.}
+\item All calibration products should be constructed from a DM Task in
+  the \verb|cp_pipe| package, and have the input data and
+  configuration parameters persisted with the calibration itself.
+  This guarantees that any product can be reconstructed in the future.
+  Exceptions will exist for external or lab created calibrations,
+  although the majority of used calibrations will likely be produced
+  directly.
+\item These provenance, config, and final product files may be saved
+  to an \verb|obs_CAMERA_data| package in git for convenience.  This
+  also allows these files to be migrated to other Butler repositories
+  in a consistent fashion.  Making these products human readable
+  allows the tracking of value changes with the evolution of the set
+  of exported products.
+\item All calibration products are accepted for use and have a date
```

Comment:
Does all accepting for use (certification) have to be manual? What about beginning-of-night or during-night calibrations?  What about using raw calibrations to do prompt processing of subsequent raw calibrations?  What about Commissioning where raw calibrations are used for subsequent science-like images?

Reply:
I know that during the OCPS discussions @RobertLuptonTheGood required that at the summit a file would automatically be calibrated with the data taken prior to it. The implication was that this would be automatic and not involve the observer doing a certification step but maybe I misunderstood.

Reply:
Applying calibrations definitely needs to be automatic.  My understanding is that the CPP team does not want us to _assume_ any calibration certification operation could be automatic, but is open to adding automated procedures in the future, on a case-by-case basis, when that automation can be demonstrated to be safe.

Reply:
I expect that data taken at the summit will be handled like any other data:  it's calibrated using the current certified calibrations.  If we need an auto-certification step on the mountain that'd be OK.

Reply:
One other point that fits here is that I had planned that the to-be-written `cp_validate` package would be usable on during-night calibrations to provide some set of metrics to show that the current certified calibration is consistent with new data.